datadog/resource_datadog_service_level_objective.go

```diff
@@ -0,0 +1,489 @@
+			"target":    threshold.GetTarget(),
+			"warning":   threshold.GetWarning(),
+		}
+		if targetDisplay, ok := threshold.GetTargetDisplayOk(); ok {
+			t["target_display"] = targetDisplay
+		}
+		if warningDisplay, ok := threshold.GetWarningDisplayOk(); ok {
+			t["warning_display"] = warningDisplay
+		}
+		thresholds = append(thresholds, t)
+	}
+
+	tags := make([]string, 0)
+	for _, s := range slo.Tags {
+		tags = append(tags, s)
+	}
+	sort.Strings(tags)
+
+	d.Set("name", slo.GetName())
+	d.Set("description", slo.GetDescription())
+	d.Set("type", slo.GetType())
+	d.Set("tags", tags)
+	d.Set("thresholds", thresholds)
+	switch slo.GetType() {
+	case datadog.ServiceLevelObjectiveTypeMonitor:
+		// monitor type
+		if len(slo.MonitorIDs) > 0 {
+			sort.Ints(slo.MonitorIDs)
+			d.Set("monitor_ids", slo.MonitorIDs)
+		}
+		if ms, ok := slo.GetMonitorSearchOk(); ok {
+			d.Set("monitor_search", ms)
+		}
+		sort.Strings(slo.Groups)
+		d.Set("groups", slo.Groups)
+	default:
+		// metric type
+		query := make(map[string]interface{})
+		q := slo.GetQuery()
+		query["numerator"] = q.GetNumerator()
+		query["denominator"] = q.GetDenominator()
+		d.Set("query", []map[string]interface{}{query})
+	}
+
+	return nil
+}
+
+func resourceDatadogServiceLevelObjectiveUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*datadog.Client)
+	slo := &datadog.ServiceLevelObjective{
+		ID: datadog.String(d.Id()),
+	}
+
+	if attr, ok := d.GetOk("name"); ok {
+		slo.SetName(attr.(string))
+	}
+
+	if attr, ok := d.GetOk("description"); ok {
+		slo.SetDescription(attr.(string))
+	}
+
+	if attr, ok := d.GetOk("type"); ok {
+		slo.SetType(attr.(string))
+	}
+
+	switch slo.GetType() {
+	case datadog.ServiceLevelObjectiveTypeMonitor:
+		// monitor type
+		if attr, ok := d.GetOk("monitor_ids"); ok {
+			s := make([]int, 0)
+			for _, v := range attr.(*schema.Set).List() {
+				s = append(s, v.(int))
+			}
+			sort.Ints(s)
+			slo.MonitorIDs = s
+		}
+		if attr, ok := d.GetOk("monitor_search"); ok {
+			slo.SetMonitorSearch(attr.(string))
+		}
+		if attr, ok := d.GetOk("groups"); ok {
+			s := make([]string, 0)
+			for _, v := range attr.(*schema.Set).List() {
+				s = append(s, v.(string))
+			}
+			sort.Strings(s)
+			slo.Groups = s
+		}
+	default:
+		// metric type
+		if attr, ok := d.GetOk("query"); ok {
+			if query, ok := attr.([]map[string]interface{}); ok && len(query) == 1 {
+				slo.SetQuery(datadog.ServiceLevelObjectiveMetricQuery{
+					Numerator:   datadog.String(query[0]["numerator"].(string)),
+					Denominator: datadog.String(query[0]["denominator"].(string)),
+				})
+			}
+		}
+	}
+
+	if attr, ok := d.GetOk("tags"); ok {
+		s := make([]string, 0)
+		for _, v := range attr.(*schema.Set).List() {
+			s = append(s, v.(string))
+		}
+		sort.Strings(s)
+		slo.Tags = s
+	}
+
+	if attr, ok := d.GetOk("thresholds"); ok {
+		sloThresholds := make(datadog.ServiceLevelObjectiveThresholds, 0)
+		thresholds := attr.([]map[string]interface{})
```

Comment:
TF 0.11 gives me a panic here on subsequent updates. Likely related to my comment on the diffsuppress function.